rest_lights.cpp

```diff
@@ -1974,101 +1974,135 @@ int DeRestPluginPrivate::setWindowCoveringState(const ApiRequest &req, ApiRespon
 int DeRestPluginPrivate::setTuyaDeviceState(const ApiRequest &req, ApiResponse &rsp, TaskItem &taskRef, QVariantMap &map)
 {
     QString id = req.path[3];
+    
     bool targetOn = false;
-
-    if (map.contains("on"))
+    bool hasOn = false;
+    bool hasBri = false;
+    uint targetBri = 0;
+    
+    bool ok = false;
+    
+    //Parse all parameters
+    for (QVariantMap::const_iterator p = map.begin(); p != map.end(); p++)
     {
-        if (map["on"].type() == QVariant::Bool)
+        if (p.key() == "bri" && R_GetProductId(taskRef.lightNode).startsWith(QLatin1String("Tuya_DIMSWITCH")))
         {
-            bool ok = false;
-            qint8 button = DP_IDENTIFIER_BUTTON_1;
-            QByteArray data;
-
-            targetOn = map["on"].toBool();
-
-            //Retreive Fake endpoint, and change button value
-            uint8_t ep = taskRef.lightNode->haEndpoint().endpoint();
-            if (ep == 0x02) { button = DP_IDENTIFIER_BUTTON_2; }
-            if (ep == 0x03) { button = DP_IDENTIFIER_BUTTON_3; }
-
-            //Use only the first endpoint for command
-            taskRef.req.setDstEndpoint(0x01);
+            if (map[p.key()].type() == QVariant::Double)
+            {
+                targetBri = map["bri"].toUInt(&ok);
+                if (ok)
+                {
+                    hasBri = true;
+                }
+            }
 
-            DBG_Printf(DBG_INFO, "Tuya debug 10: EP: %d ID : %s\n", ep, qPrintable(id));
+            if (!hasBri)

```

Comment:
Cool I like that now all unsupported parameters are error reported.

The `!hasBri` branch should check the valid range to report this error as well.

```
targetBri = map["bri"].toUInt(&ok);
if (ok && targetBri <= 0xFF)
{
    hasBri = true;
}
```

Reply:
done